test/MUXControlsTestApp/VisualTreeTestBase.cs

```diff
@@ -0,0 +1,360 @@
+﻿using Common;
+using System;
+using System.Threading.Tasks;
+using Windows.Storage;
+using System.IO;
+using MUXControls.TestAppUtils;
+using MUXControlsTestApp.Utilities;
+using System.Text;
+using System.Collections.Generic;
+using Windows.UI.Xaml.Markup;
+using DiffPlex.DiffBuilder;
+using DiffPlex.DiffBuilder.Model;
+
+#if USING_TAEF
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+using WEX.Logging.Interop;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+#endif
+
+using PlatformConfiguration = Common.PlatformConfiguration;
+
+namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests
+{
+    public enum Theme
+    {
+        Dark,
+        Light,
+        All
+    }
+
+    /// <summary>
+    /// TestBaseClass helps to compare visualtree.
+    /// Don't add <c>[TestInitialize]</c> in your test case, instead, just override <see cref="OnTestInitialized"/>
+    ///  <see cref="VerifyVisualTree"/> helps to compare the visualtree with a 'master' file.
+    /// Master filename for testcase would [testclass]_[testname]_[theme].xml or [testclass]_[testname]_[theme]-[apiversion].xml
+    /// </summary>
+    public class VisualTreeTestBase
+    {
+        public const string LogMasterFileRuntimeParameterName = "LogMasterFile";
+
+        public VisualTreeDumper.IFilter VisualTreeDumpFilter { get; set; }
+        public VisualTreeDumper.IPropertyValueTranslator PropertyValueTranslator { get; set; }
+
+        public TestContext TestContext { get; set; }
+
+        public bool ShouldLogMasterFile { get; set; }
+
+        public string TestCaseName { get; private set; }
+
+        // To avoid filename conflict for master file, we use the format of [testclass]_[testname] as prefix by default.
+        // For example, Windows.UI.Xaml.Tests.MUXControls.ApiTests.NavigationViewVisualTreeTests.VerifyVisualTreeForNavView
+        // The prefix is NavigationViewVisualTreeTests_VerifyVisualTreeForNavView
+        public string MasterFileNamePrefix { get; set; }
```

Comment:
Why not just make the file name same as namespace.classname.methodname ?

Reply:
Filename will be too long. New implementation will ask you to input a uniq testprefix